Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4))

**Describe the bug**

Error during docker image build on apple M1:

```
#21 50.09 /opt/openmpi/bin/mpifort -DPACKAGE_NAME=\"GFDL\ FMS\ Library\" -DPACKAGE_TARNAME=\"FMS\" -DPACKAGE_VERSION=\"2021.04.0\" -DPACKAGE_STRING=\"GFDL\ FMS\ Library\ 2021.04.0\" -DPACKAGE_BUGREPORT=\"gfdl.climate.model.info@noaa.gov\" -DPACKAGE_URL=\"https://www.gfdl.noaa.gov/fms\" -DPACKAGE=\"FMS\" -DVERSION=\"2021.04.0\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_MPI_H=1 -DHAVE_NETCDF_H=1 -DHAVE_GETTID=1 -DHAVE_SCHED_GETAFFINITY=1 -DHAVE_MOD_MPI=1 -DHAVE_MOD_NETCDF=1 -DHAVE_CRAY_POINTER=1 -DHAVE_QUAD_PRECISION=1 -DHAVE_INTERNAL_NML=1 -Duse_netCDF=1 -Duse_libMPI=1 -I.  -I../include -I../mosaic -I/opt/netcdf/include -fopenmp -I. -J../.mods -I/opt/netcdf/include -ffree-form -ffree-line-length-none -fdefault-real-8 -fcray-pointer -fopenmp -c mosaic2.F90
#21 50.13 mosaic2.F90:512:56:
#21 50.13 
#21 50.13   503 |            call get_grid_great_circle_area( nlon, nlat, lon, lat, area)
#21 50.13       |                                                        2
#21 50.13 ......
#21 50.13   512 |            call get_grid_great_circle_area( nlon, nlat, lon, lat, area)
#21 50.13       |                                                        1
#21 50.13 Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4)).
#21 50.13 mosaic2.F90:462:43:
#21 50.13 
#21 50.13   453 |            call get_grid_area( nlon, nlat, lon, lat, area)
#21 50.13       |                                           2
#21 50.13 ......
#21 50.13   462 |            call get_grid_area( nlon, nlat, lon, lat, area)
#21 50.13       |                                           1
#21 50.13 Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4)).
#21 50.13 make[1]: Leaving directory '/FMS/mosaic2'
#21 50.13 make[1]: *** [Makefile:701: mosaic2_mod.mod] Error 1
#21 50.13 make: *** [Makefile:675: all-recursive] Error 1
------
executor failed running [/bin/sh -c git clone https://github.com/NOAA-GFDL/FMS.git  && cd FMS  && git checkout $FMS_VERSION  && autoreconf -i  && CPPFLAGS="-I${NETCDF_DIR}/include"     FCFLAGS="-I${NETCDF_DIR}/include"     LDFLAGS="-L${NETCDF_DIR}/lib"     ./configure --prefix=${FMS_DIR}  && make -j${MAKEJOBS} && make install]: exit code: 2
```

**To Reproduce**
Build FMS (both 2021.04 and main branch) in Fedora on apple M1 arm/aarch64, building docker image

**Expected behavior**
Successful build.

**System Environment**
Describe the system environment, include:
- OS: 
Fedora for M1/ARM64
5.10.76-linuxkit
- Compiler(s): Type and version [e.g. Intel 19.1]
gcc (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)
GNU Fortran (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)
- MPI type, and version (e.g. MPICH, Cray MPI, openMPI)
ARG OPENMPI_VERSION="v4.1.0"
- netCDF Version: For both C and Fortran
ARG NETCDF_VERSION="4.8.1"
ARG NETCDFF_VERSION="4.5.4"
- Configure options: Any additional flags, or macros passed to configure

**Additional context**
I can provide the Dockerfile if requested.

--- a/mosaic2/mosaic2.F90
+++ b/mosaic2/mosaic2.F90
@@ -450,7 +450,7 @@ end function transfer_to_model_index
        type is (real(r4_kind))
          select type(area)
          type is (real(r4_kind))
-           call get_grid_area( nlon, nlat, lon, lat, area)
+           call get_grid_area( nlon, nlat, real(lon, r8_kind), real(lat, r8_kind), real(area, r8_kind))
            valid_types = .true.
          end select
        end select
@@ -500,7 +500,7 @@ end function transfer_to_model_index
        type is (real(r4_kind))
          select type(area)
          type is (real(r4_kind))
-           call get_grid_great_circle_area( nlon, nlat, lon, lat, area)
+           call get_grid_great_circle_area( nlon, nlat, real(lon, r8_kind), real(lat, r8_kind), real(area, r8_kind))
            valid_types = .true.
          end select
        end select

